app/routes/_layout+/kits.$kitId.give-custody.tsx

```diff
@@ -0,0 +1,269 @@
+import { AssetStatus, KitStatus } from "@prisma/client";
+import { json, redirect } from "@remix-run/node";
+import type { ActionFunctionArgs, LoaderFunctionArgs } from "@remix-run/node";
+import { Form, useActionData, useNavigation } from "@remix-run/react";
+import { z } from "zod";
+import DynamicSelect from "~/components/dynamic-select/dynamic-select";
+import { UserIcon } from "~/components/icons/library";
+import { Button } from "~/components/shared/button";
+import { db } from "~/database/db.server";
+import { createNote } from "~/modules/asset/service.server";
+import { getKit } from "~/modules/kit/service.server";
+import { getUserByID } from "~/modules/user/service.server";
+import styles from "~/styles/layout/custom-modal.css?url";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+import { makeShelfError, ShelfError } from "~/utils/error";
+import { isFormProcessing } from "~/utils/form";
+import {
+  assertIsPost,
+  data,
+  error,
+  getCurrentSearchParams,
+  getParams,
+  parseData,
+} from "~/utils/http.server";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
+import { stringToJSONSchema } from "~/utils/zod";
+
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { kitId } = getParams(params, z.object({ kitId: z.string() }), {
+    additionalData: {
+      userId,
+    },
+  });
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.kit,
+      action: PermissionAction.update,
+    });
+
+    const kit = await getKit({
+      id: kitId,
+      organizationId,
+      extraInclude: {
+        assets: { select: { status: true } },
+      },
+    });
+
+    if (kit.custody) {
+      return redirect(`/kits/${kitId}`);
+    }
+
+    /**
+     * If any asset is not available in a kit,
+     * then a kit cannot be assigned a custody
+     */
+    const someUnavailableAsset = kit.assets.some(
```

Comment:
I understand what you are doing here and I think its important to check this one the server, but what I usually do is I also disabled the actual give-custody button. Thats how I do it on asset. Do we have the button disabled here?

Reply:
Yes, I have also disabled the button on `give-custody` button